### PR TITLE
Constructors required by GCC 4.9 when compiling for Android

### LINKED ISSUE #205: master broken on Android
```
/Users/cm/Realm/realm-object-store/src/impl/realm_coordinator.cpp: In member function 'bool {anonymous}::IncrementalChangeInfo::advance_incremental(realm::VersionID)':
/Users/cm/Realm/realm-object-store/src/impl/realm_coordinator.cpp:428:45: error: no matching function for call to 'std::vector<realm::_impl::TransactionChangeInfo>::push_back(<brace-enclosed initializer list>)'
                 std::move(m_current->lists)});
                                             ^
/Users/cm/Realm/realm-object-store/src/impl/realm_coordinator.cpp:428:45: note: candidates are:
In file included from /usr/local/Cellar/android-ndk-r10e/r10e/sources/cxx-stl/gnu-libstdc++/4.9/include/vector:64:0,
                 from /Users/cm/Realm/realm-object-store/src/schema.hpp:23,
                 from /Users/cm/Realm/realm-object-store/src/shared_realm.hpp:22,
                 from /Users/cm/Realm/realm-object-store/src/impl/realm_coordinator.hpp:22,
                 from /Users/cm/Realm/realm-object-store/src/impl/realm_coordinator.cpp:19:
/usr/local/Cellar/android-ndk-r10e/r10e/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/stl_vector.h:1038:7: note: void std::vector<_Tp, _Alloc>::push_back(const value_type&) [with _Tp = realm::_impl::TransactionChangeInfo; _Alloc = std::allocator<realm::_impl::TransactionChangeInfo>; std::vector<_Tp, _Alloc>::value_type = realm::_impl::TransactionChangeInfo]
       push_back(const value_type& __x)
       ^
/usr/local/Cellar/android-ndk-r10e/r10e/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/stl_vector.h:1038:7: note:   no known conversion for argument 1 from '<brace-enclosed initializer list>' to 'const value_type& {aka const realm::_impl::TransactionChangeInfo&}'
/usr/local/Cellar/android-ndk-r10e/r10e/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/stl_vector.h:1057:7: note: void std::vector<_Tp, _Alloc>::push_back(std::vector<_Tp, _Alloc>::value_type&&) [with _Tp = realm::_impl::TransactionChangeInfo; _Alloc = std::allocator<realm::_impl::TransactionChangeInfo>; std::vector<_Tp, _Alloc>::value_type = realm::_impl::TransactionChangeInfo]
       push_back(value_type&& __x)
       ^
/usr/local/Cellar/android-ndk-r10e/r10e/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/stl_vector.h:1057:7: note:   no known conversion for argument 1 from '<brace-enclosed initializer list>' to 'std::vector<realm::_impl::TransactionChangeInfo>::value_type&& {aka realm::_impl::TransactionChangeInfo&&}'
make[2]: *** [src/CMakeFiles/realm-object-store.dir/impl/realm_coordinator.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/realm-object-store.dir/all] Error 2
make: *** [all] Error 2
```

Reproduce:
Follow the instructions for building for Android. `make` will crash with the above.

### PR DESCRIPTION
**Pull request by cmelchior**
Fixes #205 

@bdash @tgoyne

### CODE REVIEW

File: src/impl/collection_notifier.hpp [hunk 1]
```diff
@@ -49,6 +49,16 @@ struct TransactionChangeInfo {
     std::vector<ListChangeInfo> lists;
     std::vector<CollectionChangeBuilder> tables;
     bool track_all = false;
+
+    // Required by GCC 4.9
+    TransactionChangeInfo() {}
+    TransactionChangeInfo(std::vector<bool> table_mods,
+                          std::vector<bool> table_moves,
+                          std::vector<ListChangeInfo> ls)
+            : table_modifications_needed(table_mods),
```
Comment (tgoyne) [inline src/impl/collection_notifier.hpp]: These all should be moved rather than copied (i.e. `table_modifications_needed(std::move(table_mods))` etc.).

**Comment by bdash** [comment]
Any reason for the arguments to have abbreviated names?

File: src/impl/collection_notifier.hpp [hunk 2]
```diff
@@ -49,6 +49,16 @@ struct TransactionChangeInfo {
     std::vector<ListChangeInfo> lists;
     std::vector<CollectionChangeBuilder> tables;
     bool track_all = false;
+
+    // Required by GCC 4.9
+    TransactionChangeInfo() {}
+    TransactionChangeInfo(std::vector<bool> table_mods,
+                          std::vector<bool> table_moves,
+                          std::vector<ListChangeInfo> ls)
+            : table_modifications_needed(std::move(table_mods)),
```
Comment (bdash) [inline src/impl/collection_notifier.hpp]: This indentation looks too deep.

File: src/impl/collection_notifier.hpp [hunk 3]
```diff
@@ -49,6 +49,16 @@ struct TransactionChangeInfo {
     std::vector<ListChangeInfo> lists;
     std::vector<CollectionChangeBuilder> tables;
     bool track_all = false;
+
+    // Required by GCC 4.9
```
Comment (kneth) [inline src/impl/collection_notifier.hpp]: Can we guard this so only gcc 4.9 sees it? See for example https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html

Comment (cmelchior) [inline src/impl/collection_notifier.hpp]: Adding guard to stay consistent with e.g `property.hpp`

**Pull request merged**